src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -367,51 +385,112 @@ namespace ts.server.typingsInstaller {
             }
         }
 
-        private watchFiles(projectName: string, files: string[]) {
+        private watchFiles(projectName: string, files: string[], projectRootPath: Path) {
             if (!files.length) {
                 // shut down existing watchers
                 this.closeWatchers(projectName);
                 return;
             }
 
             let watchers = this.projectWatchers.get(projectName);
+            const toRemove = createMap<FileWatcher>();
             if (!watchers) {
                 watchers = createMap();
                 this.projectWatchers.set(projectName, watchers);
             }
+            else {
+                copyEntries(watchers, toRemove);
+            }
 
-            watchers.isInvoked = false;
             // handler should be invoked once for the entire set of files since it will trigger full rediscovery of typings
+            watchers.isInvoked = false;

```

Comment:
This is probably a dumb question, but isn't `watchers` a map?  Why would it have an `isInvoked` property?

Reply:
its a expando on map.. This is so that we arent catching incorrect local variable. the state recides on the map (more on this https://github.com/Microsoft/TypeScript/pull/23438#discussion_r182237104)